tests/lit-tests/func_template_nontype_5.ispc

```diff
@@ -0,0 +1,38 @@
+// Basic test for templates with enum template parameters.
+// RUN: %{ispc}  %s --emit-llvm-text --target=host --nostdlib -o - | FileCheck %s
+
+// CHECK-LABEL: define {{.*}} @enum_foo___uniCunenum_5B_Foo_5D_ZERO___uni
+
+// CHECK-LABEL: define {{.*}} @bar_uniform___uni
+// CHECK: call i32 @enum_foo___uniCunenum_5B_Foo_5D_TWO___uni
+
+// CHECK-LABEL: define linkonce_odr i32 @enum_foo___uniCunenum_5B_Foo_5D_TWO___uni
+
+// CHECK-LABEL: define {{.*}} @bar_varying___uni
+// CHECK: call <{{[0-9]*}} x i32> @enum_foo___vyiCunenum_5B_Foo_5D_TWO___uni
+
+// CHECK-LABEL: define linkonce_odr <{{[0-9]*}} x i32> @enum_foo___vyiCunenum_5B_Foo_5D_TWO___uni
+
+enum Foo {
+    ZERO,
+    ONE,
+    TWO
+};
+
+template<typename T, Foo N>
+noinline T enum_foo(uniform int k) {
+    return k + N;
+}
+
+template<>
+noinline uniform int enum_foo<uniform int, ZERO>(uniform int k) {
+    return k;
+}
+
+uniform int bar_uniform(uniform int k) {
+    return enum_foo<uniform int, TWO>(k);
+}
+
+int bar_varying(uniform int k) {
+    return enum_foo<varying int, TWO>(k);
+}
```

Comment:
I would add test like this here:
```ispc
template<Foo N>
uniform int func() {
    switch (N) {
        case ZERO:
            return 0;
        case ONE:
            return 1;
        case TWO:
            return 2;
    }
    return 42;
}

uniform int func_1() { return func<ONE>(); } // ideally to check that it is just ret i32 1
```

Reply:
Added.

Reply:
I see that templated function are not inlined by some reason:
```llvm-ir
; Function Attrs: nounwind
define i32 @bar_uniform___uni(i32 %k, <8 x i32> %__mask) local_unnamed_addr #1 {
allocas:
  %calltmp = tail call i32 @enum_foo___uniCunenum_5B_Foo_5D_TWO___uni(i32 %k, <8 x i32> %__mask)
  ret i32 %calltmp
}

; Function Attrs: noinline nounwind
define linkonce_odr i32 @enum_foo___uniCunenum_5B_Foo_5D_TWO___uni(i32 %k, <8 x i32> %__mask) local_unnamed_addr #2 {
allocas:
  ret i32 2
}
```

Reply:
Yes, they are declared with `noinline` keyword.

Reply:
Ah, I see now. But why do we need them? My initial point was to test exactly this:
```ispc
// CHECK-LABEL: @bar_uniform___uni
// CHECK-NEXT: allocas:
// CHECK-NEXT: ret i32 2
uniform int bar_uniform(uniform int k) {
    return enum_foo<uniform int, TWO>(k);
}
```